src/ham/hamd/hamd.h

```diff
@@ -0,0 +1,115 @@
+// Host Account Management
+#ifndef HAMD_H
+#define HAMD_H
+
+#include <dbus-c++/dbus.h>          // DBus::Connection
+#include <glib.h>                   // gint, gpointer
+#include "timer.h"                  // gtimer_c
+#include "../shared/dbus-address.h" // DBUS_BUS_NAME_BASE
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-but-set-variable"   /* SUPPRESS: warning: variable 'ri' set but not used [-Wunused-but-set-variable] */
+#include "../shared/org.SONiC.HostAccountManagement.dbus-adaptor.h"
+#pragma GCC diagnostic pop
+
+class hamd_config_c
+{
+public:
+    hamd_config_c(int argc, char **argv);
+
+    /**
+     * @brief Read configuration and update hamd_config_c object
+     */
+    void reload();
+
+    uid_t uid_fit_into_range(uint64_t hash) const
+    {
+        return (uid_t)((hash % sac_uid_range_m) + sac_uid_min_m);
+    }
+
+    /**
+     * @brief return the shell program to assign to new users.
+     */
+    const std::string & shell() const {return shell_m;}
+
+    /**
+     * @brief return the command to execute to create a user's certificates.
+     */
+    std::string certgen_cmd(const std::string & user_r, const std::string & certdir_r) const;
+
+private:
+    static const  gint  poll_period_sec_default_m = 30;
+    static const  gint  sac_uid_min_default_m     = 5000;  // System-Assigned IDs will be in the
+    static const  gint  sac_uid_max_default_m     = 59999; // range [sac_uid_min_m..sac_uid_max_m]
+    static const  bool  tron_default_m            = false;
+    const gchar       * conf_file_default_pm      = "/etc/sonic/hamd/config";
+    std::string         certgen_cmd_default_m     = "/usr/bin/openssl req -newkey rsa:2048 -nodes -keyout $CERTDIR/key.pem -subj \"/O=SONiC/OU=CLI/CN=$USERNAME\" | /usr/bin/openssl x509 -req -days 365000 -out $CERTDIR/certificate.pem -CA /root/cli-ca/certificate.pem -CAkey /root/cli-ca/key.pem -CAcreateserial -sha256";
+    std::string         shell_default_m           = "/usr/bin/sonic-cli";
```

Comment:
Need to check on default shell. I think we decided to go with defaulting to Bash for now instead of into clish. Not sure what "sonic-cli" is

Reply:
sonic-cli is the script to start clish inside the container